extensions/rocksdb-repos/ProvenanceRepository.h

```diff
@@ -72,7 +72,7 @@ class ProvenanceRepository : public core::Repository, public std::enable_shared_
     if (running_)
       return;
     running_ = true;
-    thread_ = std::thread(&ProvenanceRepository::run, shared_from_this());
```

Comment:
I can confirm that this fixes the ProvenanceRepository crash.  Are the tests going to be in this pull request, or in a separate one?